GiniVision/Classes/GiniImageDocument.swift

```diff
@@ -17,16 +17,23 @@ final public class GiniImageDocument: NSObject, GiniVisionDocument {
     public var data:Data
     public var previewImage: UIImage?
     
+    fileprivate let metaInformationManager:ImageMetaInformationManager
+    
     /**
      Initializes a GiniImageDocument.
      
      - Parameter data: PDF data
+     - Parameter deviceOrientation: Device orientation when a picture was taken from the camera. In other cases it should be `nil`
      
      */
     
-    public init(data: Data) {
-        self.data = data
+    public init(data: Data, deviceOrientation:UIInterfaceOrientation? = nil) {
         self.previewImage = UIImage(data: data)
+        self.metaInformationManager = ImageMetaInformationManager(imageData: data, deviceOrientation:deviceOrientation)
+        self.metaInformationManager.filterMetaInformation()
```

Comment:
I know this method was there before you, but I think it's misleading. First of all, it doesn't filter anything - just adds the meta tags. Second of all, I'd remove it from the public interface altogether. You will always want the tags, so just call it internally at the end of the init.

Reply:
It makes sense, since it is only used when the manager is initialized. But, since this class needs to be refactored, I think we should keep the method name until we refactor it in order to avoid misunderstandings in the future.